### PR TITLE
Merge to Staging for v3.0.0 release

### PR DESCRIPTION
**Pull request by kevinlind**
## Description

Merge to Staging for v3.0.0 release:

Major version update for Edge Bridg for Adobe Experience Platform Mobile SDKs on Android compatible with Mobile Core 3.0.0. The current release includes the following changes:

- Sets minSDK to 21
- Sets compileSDK and targetSDK to 34
- Rewrite Gradle files in Kotlin and use aepsdk-gradle-plugin for common build and publish tasks used across AEP SDKs.
- Updates CircleCI configuration orbs and machine images.
- Updates Android Gradle Plugin to 8.2
- Updated Edge Bridge data format to enhance support for migrating to sending data via Edge Network to Adobe Analytics and reduce the effort required for mapping data to XDM.
- If you are currently using Edge Bridge version 2.x and you are mapping data to XDM in your datastream, adjustments will be necessary when adopting version 3.0.0. For more details, please review the [migration](https://developer.adobe.com/client-sdks/resources/migration/android/migrate-to-3x/) documentation.

## Types of changes

- [ ] Bug fix (non-breaking change which fixes an issue)
- [x] New feature (non-breaking change which adds functionality)
- [x] Breaking change (fix or feature that would cause existing functionality to change)

## Checklist:

- [ ] I have signed the [Adobe Open Source CLA](https://opensource.adobe.com/cla.html).
- [ ] My code follows the code style of this project.
- [ ] My change requires a change to the documentation.
- [ ] I have updated the documentation accordingly.
- [ ] I have read the **CONTRIBUTING** document.
- [ ] I have added tests to cover my changes.
- [ ] All new and existing tests passed.

### CODE REVIEW

File: .github/workflows/update-version.yml [hunk 1]
```diff
@@ -0,0 +1,32 @@
+name: Update Version
+on:
+  workflow_dispatch:
+    inputs:
+      version: 
+        description: 'New version to use for the Edge Bridge extension. Example: 3.0.0'
+        required: true
+
+      core-dependency:
+        description: '[Optional] Update Core dependency in pom.xml. Example: 3.0.0'
```
Comment (emdobrin) [inline .github/workflows/update-version.yml:10]: should pom.xml be gradle.properties ?

File: code/app/build.gradle.kts [hunk 2]
```diff
@@ -0,0 +1,70 @@
...
+    kotlinOptions {
+        jvmTarget = BuildConstants.Versions.KOTLIN_JVM_TARGET
+        languageVersion = BuildConstants.Versions.KOTLIN_LANGUAGE_VERSION
+        apiVersion = BuildConstants.Versions.KOTLIN_API_VERSION
+    }
+}
+
+dependencies {
+    implementation(project(":edgebridge"))
+    // TODO: Use 3.x versions for testing
```
Comment (emdobrin) [inline code/app/build.gradle.kts:61]: can we update to snapshot builds here

File: code/settings.gradle.kts [hunk 3]
```diff
@@ -0,0 +1,36 @@
+/*
+ * Copyright 2022 Adobe. All rights reserved.
```
Comment (emdobrin) [inline code/settings.gradle.kts:2]: new file ©️ 2024

File: code/edgebridge/build.gradle.kts [hunk 4]
```diff
@@ -0,0 +1,46 @@
...
+
+dependencies {
+    // TODO: Use 3.x versions for testing
+    // TODO: Remove -SNAPSHOT suffix after Core 3.0.0 is published
+    implementation("com.adobe.marketing.mobile:core:$mavenCoreVersion-SNAPSHOT")
+
+    // testImplementation dependencies provided by aep-library:
+    // MOCKITO_CORE, MOCKITO_INLINE
+
+    // TODO: Use 3.x versions for testing
```
Comment (emdobrin) [inline code/edgebridge/build.gradle.kts:38]: let's add 3.x snapshots here too

File: code/settings.gradle.kts [hunk 5]
```diff
@@ -0,0 +1,36 @@
...
+    repositories {
+        google()
+        mavenCentral()
+        mavenLocal()
+        maven { url = uri("https://oss.sonatype.org/content/repositories/snapshots/") }
+        maven { url = uri("https://jitpack.io") }
+    }
+}
+
+rootProject.name = "edgebridge-sdk"
```
Comment (kevinlind) [inline code/settings.gradle.kts:32]: For consistency with other project, change to `aepsdk-edgebridge-android`

File: .github/workflows/update-version.yml (hunk 1 above)
Comment (kevinlind) [reply .github/workflows/update-version.yml:10]: This is consistent with our other extensions.
https://github.com/adobe/aepsdk-edge-android/blob/staging/.github/workflows/update-version.yml#L22
https://github.com/adobe/aepsdk-analytics-android/blob/staging/.github/workflows/update-version.yml#L22

**Review by emdobrin — APPROVED** [review]

**Pull request merged**